bin/p2-exec/main.go

```diff
@@ -247,7 +274,46 @@ func cgEnter(platconf, launchableName, cgroupName string) error {
 	} else if err != nil {
 		return util.Errorf("Could not set cgroup parameters: %s", err)
 	}
-	return cg.AddPID(cgConfig.Name, 0)
+	return cg.AddPID(config.Name, 0)
+}
+
+// given ENV variable "configPath" return parameters for the new cgroup being created
+func cgGetConfig(configPath string, configKey string) (cgroups.Config, error) {
+	if config := os.Getenv(configPath); config != "" {
+		confBuf, err := ioutil.ReadFile(config)
+		if err != nil {
+			return cgroups.Config{}, err
+		}
+		if configPath == "RESOURCE_LIMITS" {
```

Comment:
given that like 80% of this function is in this if/else block, maybe it would make sense to have separate functions like `getPodCGConfig` and `getLaunchableCGConfig`.